strings/my_vsnprintf.c

```diff
@@ -749,23 +789,56 @@ size_t my_vsnprintf_ex(CHARSET_INFO *cs, char *to, size_t n,
       to= process_dbl_arg(to, end, width, d, *fmt);
       continue;
     }
-    else if (*fmt == 'd' || *fmt == 'i' || *fmt == 'u' || *fmt == 'x' || 
-             *fmt == 'X' || *fmt == 'p' || *fmt == 'o')
+    case 'd':
+    case 'i':
+    case 'u':
+    case 'x':
+    case 'X':
+    case 'o':
+    case 'p':
+    case 'M':
     {
       /* Integer parameter */
       longlong larg;
-
+      const char arg_type= *fmt;
+      my_bool suffix_e= arg_type == 'M';
       if (have_longlong)
-        larg = va_arg(ap,longlong);
-      else if (*fmt == 'd' || *fmt == 'i')
-        larg = va_arg(ap, int);
+        larg= va_arg(ap,longlong);
+      else if (arg_type == 'd' || arg_type == 'i' || suffix_e)
+        larg= va_arg(ap, int);
       else
         larg= va_arg(ap, uint);
-
-      to= process_int_arg(to, end, length, larg, *fmt, print_type);
+      if (arg_type == 'u')
+        switch (fmt[1]) // look-ahead
+        {
+        case 'E':
+          suffix_e= TRUE;
+          // fall-through
+        case 'U': // escape
+          ++fmt;
+          break;
+        }
+      if (suffix_e)
```

Comment:
I think whichever you feel more comfortable with. Personally, I'm good with how it is now.

Reply:
👌
I’ll go with the natural flow.
Since the entire file could use some drastic refactoring (as added in OP), I’ll leave this structural decision to that project.

Reply:
The recent decision on changing `%uE` to `%dE`/`%iE` made this alternate design more suitable for  `%dD` and `%iI` one way or another.

This conundrum wouldn’t exist if my job was to rewrite the whole file, ehhh 🤷.